web/src/pages/api/public/dataset-run-items.ts

```diff
@@ -23,6 +27,10 @@ export default withMiddlewares({
         metadata,
       } = body;
 
+      /**************
```

Comment:
Nit: I think instead of having these headers, we could also move each part into functions that we call in here.